analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GetTypeWithIsAssignableFrom.CSharp9.Fixed.cs

```diff
@@ -0,0 +1,20 @@
+﻿Apple apple = new();
+Fruit f = apple;
+
+var b = f is Apple;     // Compliant
+b = f is not Apple;     // Compliant
+b = apple.GetType() == typeof(int?);                // Compliant
+b = apple.GetType().IsInstanceOfType(f.GetType());  // Compliant
+b = !(f is Apple);         // Fixed
+b = !(f is Apple);         // Fixed
+b = f is Apple;     // Fixed
+b = !(f is Apple); // Fixed
+b = f as Apple is Apple { Size: 42 };   // Compliant
+b = f is Apple;        // Fixed
+b = true && (apple != null);   // Fixed
+b = !(apple != null);          // Fixed
```

Comment:
this is not great, it should be refactored well from the first , I believe - also, it would better be `apple is not null`

Reply:
to do the proper refactoring might take time, I suggest to skip doing the refactoring when having double negation

Reply:
I opened https://github.com/SonarSource/sonar-dotnet/issues/4633 for this.